clarify baseline mode behavior (#8730)
fishtown-analytics/docs.getdbt.com: website/docs/docs/fusion/new-concepts.md

Reviewer: this is the latest and greatest iteration
```suggestion
| Feature | off | baseline | strict |
|---------|-----|----------|--------|
| Go-to-definition/reference (except columns) | ✅ | ✅ | ✅ |
| Table lineage | ✅ | ✅ | ✅ |
| YAML validation | ✅ | ✅ | ✅ |
| Render + preview SQL | ✅ | ✅ | ✅ |
| Unit tests | ✅ | ✅ | ✅ |
| Detect syntax errors | ❌ | ✅ | ✅ |
| Preview CTE results | ❌ | ✅ | ✅ |
| Go-to-definition/reference (columns) | ❌ | ❌ | ✅ |
| Automatic refactor column names | ❌ | ❌ | ✅ |
| Rich column lineage | ❌ | ❌ | ✅ |
| Detect data type and function signature errors | ❌ | ❌ | ✅ |
```

#### Introspection handling in baseline mode

In `baseline` mode, all static analysis findings are warnings, not errors &mdash; your project can continue running even when the compiler flags invalid or problematic SQL. This section is a good example of why that design exists.

> Previously, with `strict` mode, the system assumed local schemas of your compiled models would be available. In `baseline` mode, we can no longer assume the full local schema is available and complete, so `baseline` uses the remote database as the source of truth &mdash; similar to dbt Core.

Author: ```suggestion
Previously, with `strict` mode, the system assumed local schemas of your compiled models would be available. In `baseline` mode, we can no longer assume the full local schema is available and complete, so `baseline` uses the remote database as the source of truth &mdash; similar to <Constant name="core"/>.
```

> #### Introspection handling in baseline mode

Reviewer: @cursoragent can you make a new PR that makes this subheading be at most level 3? it's weird to me that it sits within the LSP feature comparison heading

)
```

> This is invalid SQL. In `baseline` mode, <Constant name="fusion" /> surfaces a warning so your project can continue running while still alerting you to the issue:

Author: ```suggestion
This is invalid SQL. In `baseline` mode, <Constant name="fusion" /> displays a warning so your project can continue running while still alerting you to the issue:
```
